### PR TITLE
Move built assets from dist to lib

### PR DESCRIPTION
**Pull request by matt-allan**
My last release broke Type resolution 😢

TypeScript cannot parse conditional exports, so it's unable to resolve types for paths like `radiance-ui/lib/constants` if they're actually in `radiance-ui/dist/constants`. Using the same name for the built assets ensures TypeScript can resolve the path correctly without having to parse the conditional exports.

We _could_ build the types to `lib` and the bundles to `dist` but that seemed confusing.

See https://github.com/microsoft/TypeScript/issues/33079

### CODE REVIEW

**Comment by michaeljaltamirano** [comment]
> My last release broke Type resolution 😢
> 
> TypeScript cannot parse conditional exports, so it's unable to resolve types for paths like `radiance-ui/lib/constants` if they're actually in `radiance-ui/dist/constants`. Using the same name for the built assets ensures TypeScript can resolve the path correctly without having to parse the conditional exports.
> 
> We _could_ build the types to `lib` and the bundles to `dist` but that seemed confusing.
> 
> See [microsoft/TypeScript#33079](https://github.com/microsoft/TypeScript/issues/33079)

Do you suspect this is kind of just working around the problem of clumsy import syntax? e.g., shouldn't we probably just tear off the band-aid and omit `/lib` from the import paths and instead prefer something like `radiance-ui/constants`? This would be more similar to other libraries like `rxjs/operators`. 

FWIW the migration in our main app would be painless: we can already apply `eslint --fix` across the entire repository without being blocked by errors from old code. I can't say the same for our site repos.

**Review by michaeljaltamirano — APPROVED** [review]
Approving with the understanding that this satisfies the requirements in the short-term, and we can kick the can of more breaking changes to a future major version

**Comment by matt-allan** [comment]
@michaeljaltamirano yes I agree that would be preferable long term. If the paths were simpler we wouldn't need export map rewriting at all.

**Comment by matt-allan** [comment]
Actually we could add conditional exports to export both versions to make the migration easier, then deprecate the `lib` versions in a future release 🤔

**Pull request merged**
